File: leetcode/1768-merge-strings-alternately/python/merge-strings-alternately.py
# two pointers
def mergeAtlernately(word1, word2):
    i = 0
    j = 0
    result = []
    while i < len(word1) and j < len(word2):
        result.append(word1[i])
        result.append(word2[i])
        i += 1
        j += 1
    result.append(word1[i:])
    result.append(word2[i:])

    return "".join(result)

# one pointer
def mergeAtlernately(word1, word2):
    n = max(len(word1), len(word2))
    result = []
    for i in range(n):
        if i < len(word1):
            result.append(word1[i])
        if i < len(word2):
            result.append(word2[i])

    return "".join(result)


print(mergeAtlernately("abcd", "pqr"))

File: leetcode/1768-merge-strings-alternately/cpp/merge-strings-alternately.cpp
#include <iostream>

using namespace std;

// two pointer
string mergeAlternately(string word1, string word2) {
    int m = word1.size();
    int n = word2.size();
    string result = "";
    int i = 0;
    int j = 0;

    while (i < m || j < n) {
        if (i < m) {
            result.push_back(word1[i++]);
        }
        if (j < n) {
            result.push_back(word2[j++]);
        }
    }

    return result;
}

// one pointer
string mergeAlternately(string word1, string word2) {
    int n = max(word1.size(), word2.size());
    string result = "";
    for (int i = 0; i < n; i++) {
        if (i < word1.size()) {
            result.push_back(word1.at(i));
        }
        if (i < word2.size()) {
            result.push_back(word2.at(i));
        }
    }
    return result;
}
